lib/ansible/modules/cloud/vmware/vmware_host_sriov.py

```diff
@@ -184,117 +184,112 @@ def __init__(self, module):
             self.module.fail_json(msg="Failed to find host system.")
         self.results = {"before": {}, "after": {}, "changes": {}}
 
-    def sanitize_params(self, before, hostname, vmnic):
+    def sanitize_params(self):
+        """checks user input, raise error if input incompatible
+        :return : None
         """
-        :before     : dict, of params on target interface before changing
-        :hostname   : str, hosthame
-        :vmnic      : srt, target nic
-        :return     : dict, of params forwarded to vCenter, or raise fail in case iconsistancy
-        """
-        params_to_change = {
-            "sriovEnabled": None,
-            "numVirtualFunction": None,
-            "msg": "",
-            "change": False,
-            "changes": {},
-        }
-        # change = False
+
         if self.num_virt_func < 0:
             self.module.fail_json(msg="allowed value for num_virt_func >= 0")
-
         if self.num_virt_func == 0:
             if self.sriov_on is True:
                 self.module.fail_json(
                     msg="with sriov_on == true,  alowed value for num_virt_func > 0"
                 )
-            params_to_change.update({"sriovEnabled": False, "numVirtualFunction": 0})
+            self.sriov_on = False  # fill value, if urser not provided
 
         if self.num_virt_func > 0:
             if self.sriov_on is False:
                 self.module.fail_json(
                     msg="with sriov_on == false,  alowed value for num_virt_func is 0"
                 )
-            params_to_change.update(
-                {"sriovEnabled": True, "numVirtualFunction": self.num_virt_func}
-            )
+            self.sriov_on = True  # fill value, if urser not provided
```

Comment:
same as above, since `sriov_on` is an optional arg, looks like it can be handled with assigning a default value.